api/src/main/java/pl/kolendateam/dadcard/race/MaperListRaceToDTO.java

```diff
@@ -0,0 +1,34 @@
+package pl.kolendateam.dadcard.race;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import pl.kolendateam.dadcard.race.dto.RaceBaseDTO;
+import pl.kolendateam.dadcard.race.dto.SubRaceBaseDTO;
+import pl.kolendateam.dadcard.race.entity.Race;
+
+public class MaperListRaceToDTO {
+    public static ArrayList<RaceBaseDTO> raceToRaceBaseDTO(List<Race> races){
+
+        ArrayList<RaceBaseDTO> racesDTO = new ArrayList();
+
+        for (Race race : races) {
+            boolean skipCreateRaceDTO = false;
```

Comment:
po co ta kontrolka, jeżeli rasy i tak będą nie wpisywane przez użytkownika, a wprowadzane z listy?

Reply:
Bo jak je grupuje w takie cos
    {
        "raceName": "elf",
        "avatarRaceUrl": "address value",
        "subRaces": [
            {
                "id": 1,
                "subRacesName": "księżycowe elfy",
                "avatarUrl": "address value"
            },
            {
                "id": 2,
                "subRacesName": "sloneczny elfy",
                "avatarUrl": "address value2"
            }
        ]
    },
    to tworząć ArrayList<RaceBaseDTO > mam dwa przypadku RaceBaseDTO  o danej nazwie może już istniec i wtedy dodaje mu tylka tą subRaces a jak nie istnieje to do listy dodaje nowy RaceBaseDTO. Mogłem to ogarnąć osobną funkcą która by mi returnowała i tym samym przerywała pętle albo budowniczym. Ale to rozwiazanie było prostrze